pkg/component/controller/apiendpointreconciler_test.go

```diff
@@ -160,6 +160,49 @@ func TestReconcilerWithNoNeedForUpdate(t *testing.T) {
 	assert.Equal(t, "bar", e.ObjectMeta.Annotations["foo"])
 }
 
+func TestReconcilerWithNeedForUpdate(t *testing.T) {
```

Comment:
awesome that we introduced new test as soon with the fix.

Reply:
Together with @ncopa and @makhov we first created the test to be able to debug the issue in general. 😄